mongo/options/findoptions.go

```diff
@@ -10,88 +10,137 @@ import (
 	"time"
 )
 
-// FindOptions represent all possible options to the Find() function.
+// FindOptions represents options that can be used to configure a Find operation.
 type FindOptions struct {
-	AllowPartialResults *bool          // If true, allows partial results to be returned if some shards are down.
-	BatchSize           *int32         // Specifies the number of documents to return in every batch.
-	Collation           *Collation     // Specifies a collation to be used
-	Comment             *string        // Specifies a string to help trace the operation through the database.
-	CursorType          *CursorType    // Specifies the type of cursor to use
-	Hint                interface{}    // Specifies the index to use.
-	Limit               *int64         // Sets a limit on the number of results to return.
-	Max                 interface{}    // Sets an exclusive upper bound for a specific index
-	MaxAwaitTime        *time.Duration // Specifies the maximum amount of time for the server to wait on new documents.
-	MaxTime             *time.Duration // Specifies the maximum amount of time to allow the query to run.
-	Min                 interface{}    // Specifies the inclusive lower bound for a specific index.
-	NoCursorTimeout     *bool          // If true, prevents cursors from timing out after an inactivity period.
-	OplogReplay         *bool          // Adds an option for internal use only and should not be set.
-	Projection          interface{}    // Limits the fields returned for all documents.
-	ReturnKey           *bool          // If true, only returns index keys for all result documents.
-	ShowRecordID        *bool          // If true, a $recordId field with the record identifier will be added to the returned documents.
-	Skip                *int64         // Specifies the number of documents to skip before returning
-	Snapshot            *bool          // If true, prevents the cursor from returning a document more than once because of an intervening write operation.
-	Sort                interface{}    // Specifies the order in which to return results.
+	// If true, an operation on a sharded cluster can return partial results if some shards are down rather than
+	// returning an error. The default value is false.
+	AllowPartialResults *bool
+
+	// The maximum number of documents to be included in each batch returned by the server.
+	BatchSize *int32
+
+	// Specifies a collation to use for string comparisons during the operation. This option is only valid for MongoDB
+	// versions >= 3.4. For previous server versions, the driver will return an error if this option is used. The
+	// default value is nil, which means the default collation of the collection will be used.
+	Collation *Collation
+
+	// A string that will be included in server logs, profiling logs, and currentOp queries to help trace the operation.
+	// The default is the empty string, which means that no comment will be included in the logs.
+	Comment *string
+
+	// TODO
```

Comment:
Just noting this should be addressed before merging

Reply:
I don't even know what CursorType is.  I had to look it up.

Reply:
Whoops I meant to come back and add this. Done.